python/tk_multi_reviewsubmission/submitter.py

```diff
@@ -41,75 +41,85 @@ def submit_version(self, path_to_frames, path_to_movie, thumbnail_path, sg_publi
             version_name = version_name.replace("_", " ")
             # and capitalize
             version_name = version_name.capitalize()
-        
+
         # Create the version in Shotgun
         ctx = self.__app.context
         data = {
             "code": version_name,
             "sg_status_list": self.__app.get_setting("new_version_status"),
             "entity": ctx.entity,
             "sg_task": sg_task,
-            # Leaving these fields blank to enable correct rv playback
-            # and match LA shotgun
-            # "sg_first_frame": first_frame,
-            # "sg_last_frame": last_frame,
-            "frame_count": (last_frame-first_frame+1),
+            "sg_first_frame": first_frame,
+            "sg_last_frame": last_frame,
+            "frame_count": (last_frame - first_frame + 1),
             "frame_range": "%s-%s" % (first_frame, last_frame),
             "sg_frames_have_slate": False,
             "created_by": current_user,
             "user": current_user,
             "description": comment,
-            "sg_path_to_frames": path_to_frames,
             "sg_movie_has_slate": self.__app.get_setting("mov_has_slate"),
             "project": ctx.project,
         }
 
         if sgtk.util.get_published_file_entity_type(self.__app.sgtk) == "PublishedFile":
             data["published_files"] = sg_publishes
-        else:# == "TankPublishedFile"
+        else:  # == "TankPublishedFile"
             if len(sg_publishes) > 0:
                 if len(sg_publishes) > 1:
-                    self.__app.log_warning("Only the first publish of %d can be registered for the new version!" % len(sg_publishes))
+                    self.__app.log_warning("Only the first publish of %d can be registered "
+                                           "for the new version!" % len(sg_publishes))
                 data["tank_published_file"] = sg_publishes[0]
 
+        # use published file path as path to frames instead of input frames
+        # to enable RV 'Switch to Frames' to go to that instead
+        if len(sg_publishes) > 0:
+            if len(sg_publishes) > 1:
+                self.__app.log_warning(
+                    "Only the first publish of %d can be set as path to frames "
+                    "for the new version!" % len(sg_publishes))
+            data["sg_path_to_frames"] = sg_publishes[0]["path"]["local_path"]
```

Comment:
In the case where there is more than one `PublishedFile` entity attached, you cannot guarantee that the "first" one is the path_to_frames. It could very well be any number of other dependencies, like the nuke script or input plate images. I think this logic is probably going to need to go into tk-multi-publish2 instead and then get passed as the `path_to_frames` attribute, since at the point of tk-multi-publish2 we can know which is the correct `PublishedFile`.

Reply:
In shotgun_ingest, we convert the images from slog3 (images in the incoming directory) to ACEScg (images in the publish directory) in the reviewsubmission app. So in there, we need the original frame path in the review submission app (cannot be abstracted away in publish).
I can't think of an instance where multiple published files are passed to the submitter. Each image sequence will be a separate item for which the reviewsubmission app runs, and so it should receive only one PublishedFile. Only added this because there exists a similar check for the PublishedFile linked to version as well.
Shall I instead fall back to `path_to_frames` instead of using `sg_publishes[0]` if there are >1 published files?